en/components/grid.md

```diff
@@ -75,39 +73,127 @@ Let's turn the **autoGenerate** property off and define the columns collection i
+```
+
+The column also accepts one last template that will be used when a cell is in edit mode. As with the other column templates, the provided context object is again the cell value and the cell object itself. Of course in order to have your template show up make sure to set the `editable` property of the column to true.
+
+```html
+<igx-column dataType="number" editable="true" field="Price">
+    <ng-template igxCellEditor let-cell="cell">
+        <label for="price">
+            Enter the new price tag
+        </label>
+        <input name="price" type="number" [ngModel]="cell.value" (ngModelChange)="cell.update(convertToNumber($event))" />
+    </ng-template>
+</igx-column>
+```
+
+Make sure to check the API for the [IgxGridCellComponent](#igxgridcellcomponent) in order to get accustomed with the provided properties you can use in your templates.
+
+Each of the column templates can be changed programmatically at any point through the column object itself. For example in the code below, we have declared two templates for our user data. In our Typescript code we'll get references to the templates themselves and then based on some condition we will render the appropriate template for the column in our application.
+
+```html
+<igx-grid>
+    <!-- Column declarations -->
+</igx-grid>
+
+<ng-template #normalView let-value>
+    <div class="user-details">{{ val }}</div>
+    <user-details-component></user-details-component>
+</ng-template>
+
+<ng-template #smallView let-value>
+    <div class="user-details-small">{{ val }}</div>
+</ng-template>
+```
 
 ```typescript
-public initColumns(event: IgxGridColumnInitEvent) {
-    const column: IgxColumnComponent = event.column;
+@ViewChild("normalView", { read: TemplateRef })
+public normalView: TemplateRef<any>;
+
+@ViewChild("smallView", { read: TemplateRef })
+public smallView: TemplateRef<any>;
+
+....
+
+const column = this.grid.getColumnByName("User");
+// Return the appropriate template based on some conditiion.
+// For example saved user settings, viewport size, etc.
+column.bodyTemplate = this.smallView;
+```
+
+Column properties can also be set in code in the **initColumns** event which is emitted when the columns are initialized in the grid.
+
+```typescript
+public initColumns(column: IgxGridColumn) {
+    const column: IgxColumnComponent = column;
     if (column.field === 'ProductName') {
-      column.filtering = true;
-      column.sortable = true;
-      column.editable = true;
+        column.filterable = true;
+        column.sortable = true;
+        column.editable = true;
     }
 }
 ```
 
-The code above will make the **ProductName** column sortable, filterable, and editable and will instantiate the corresponding features UI (like inputs for editing, save dialogs, etc.).
-
-<div class="divider--half"></div>
+The code above will make the **ProductName** column sortable, filterable, and editable and will instantiate the corresponding features UI (like inputs for editing, etc.).
```

Comment:
The code above will make the **ProductName** column **filterable**, sortable, and editable, and will... (keep the order the same as in the code example).